prefect_aws/s3.py

```diff
@@ -223,3 +228,215 @@ async def example_s3_list_objects_flow():
         page_iterator = page_iterator.search(f"{jmespath_query} | {{Contents: @}}")
 
     return await to_thread.run_sync(_list_objects_sync, page_iterator)
+
+
+class S3Bucket(ReadableFileSystem, WritableFileSystem):
+
+    """
+    Block used to store data using AWS S3 or S3-compatible object storage like MinIO.
+
+    Args:
+        bucket_name: Name of your bucket.
+        aws_credentials: A block containing your credentials (choose this
+            or minio_credentials).
+        minio_credentials: A block containing your credentials (choose this
+            or aws_credentials).
+        basepath: Used when you don't want to read/write at base level.
+        endpoint_url: Used for non-AWS configuration. When unspecified,
+            defaults to AWS.
+
+    Example:
+        Load stored S3Bucket configuration:
+        ```python
+        from prefect_aws import S3Bucket
+
+        s3bucket_block = S3Bucket.load("BLOCK_NAME")
+        ```
+    """
+
+    # change
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/uPezmBzEv4moXKdQJ3YyL/a1f029b423cf67f474d1eee33c1463d7/pngwing.com.png?h=250"  # noqa
+    _block_type_name = "S3 Bucket"
+
+    bucket_name: str
+    minio_credentials: Optional[MinIOCredentials]
+    aws_credentials: Optional[AwsCredentials]
+    basepath: Optional[Path]
+    endpoint_url: Optional[str]
```

Comment:
Should this be optional if only minio uses it

Reply:
> I think we should still keep the entry point, just not **init** updating it to: `"S3Bucket = prefect_aws.s3"`
> 
> Plus a few other minor fixes, and I think it should be good to go! Thanks!

[Add s3bucket to entry_point](https://github.com/PrefectHQ/prefect-aws/pull/47/commits/2578d6cc2504f16dc16b85ac4d51b932d8a25239)

Reply:
> Should this be optional if only minio uses it

endpoint_url is optional. I'm not sure I follow?